API-Samples/run_all_samples.py

```diff
@@ -11,13 +11,20 @@
     else:
         sys.exit("Cannot find samples in Debug or Release, have they been built?")
 else:
-    directory = os.cwd()
-    
-for file in os.listdir(directory):
-    if file.endswith(".exe"):
-        sample = os.path.join(directory, file)
-        if "VK_LAYER_PATH" not in os.environ:
-            if file == "enable_validation_with_callback.exe" or file == "validation_cache.exe":
-                continue
-        print("Running: ", sample)
-        subprocess.check_call(sample)
+    directory = os.getcwd()
+
+samples = []
+samplesdir = os.path.join("..", "..", "API-Samples")
+for root, dir, files in os.walk(samplesdir):
+    for file in files:
+        if file.endswith(".cpp") and "utils" not in root and "android" not in root:
+            samples.append(os.path.splitext(file)[0])
+
+samples.sort()
+for sample in samples:
+    executable = os.path.join(directory, sample)
+    if "VK_LAYER_PATH" not in os.environ:
+        if sample == "enable_validation_with_callback" or sample == "validation_cache":
+            continue
```

Comment:
This works, but is a bit unclear - I had to think about why those two samples would require `VK_LAYER_PATH`.  Suggest:
```
samples_requiring_validation_layer = ["enable_validation_with_callback", "validation_cache"]
for sample in samples:
    if "VK_LAYER_PATH" not in os.environ and sample in samples_requiring_validation_layer:
        print "Skipping {} because it requires validation layers".format(sample)
        continue
```